### ISSUE TITLE
allow customization of catalog and collection filenames

### ISSUE THREAD

**Issue by matthewhanson** [sat-utils/sat-stac#36]
Currently catalogs and collections are all called 'catalog.json'. User should be able to change default behavior, and default for collections is that they be named 'collection.json' rather than 'catalog.json'

**Comment by matthewhanson** [comment]
This seems to be mostly a non-issue.  The only times a filename is assumed is when adding a new sub-catalog (or collection) or when adding an item to a collection. In those cases, I think it is ok to assume [STAC Best Practices](https://github.com/radiantearth/stac-spec/blob/dev/catalog-spec/catalog-best-practices.md). 

With Best Practices a collection JSON file should be named collection.json rather than catalog.json which sat-stac currently implements. Since Best Practices will be released with STAC 0.7.0, this issue is moved to milestone 0.2.0 and should include:

- Rename collection JSON file to collection.json
- Review current implementation of `Collection.add_item()` to ensure it is implementing Best Practices.
- Optionally, provide two new envvars `SATUTILS_CATALOG_NAME` and `SATUTILS_COLLECTION_NAME` to specify the names of the Catalogs and Collections. This should be implemented using a [config.py file like in sat-search](https://github.com/sat-utils/sat-search/blob/master/satsearch/config.py).

**Comment by jbants** [comment]
Would a flag on `Catalog.add_catalog()`to identify the child as a collection rather than a catalog work?
ex.
```
cat_json = {
    "stac_version": "0.7.0",
    "id": "mycat",
    "description": "A child catalog of my shiny new STAC catalog",
    "root": "http://nowhere.nowhere"
}
mycat = Catalog.create(**cat_json)
mycat.save_as('/Users/james/PycharmProjects/Prescient/sat-utils/sat-stac/test/catalog.json')

breed_json = {
    "stac_version":"0.7.0",
    "id": "mybreed",
    "description": "A child collection of my shiny new STAC catalog",
    "license":"MIT",
    "extent":[],
    "links":{}
}
breed = Collection.create(**breed_json)

mycat.add_catalog(breed, collection=True)
```
